### PR TITLE
unified logic for foldable stuff

### PR DESCRIPTION
**Pull request by fterrier**
so we can use foldable javascript logic through the app

refactored the app so the reports/survey and planning tool use the same mechanism

### CODE REVIEW

File: grails-app/views/survey/_menu.gsp [hunk 1]
```diff
@@ -17,11 +17,11 @@
 					<span class="program-status-closed program-status ${enteredProgram.displayedStatus!='closed'?'hidden':''}"></span>
 				</span>
 			</a>
-			<ul class="survey-section">
+			<ul class="js_foldable-container foldable-container">
 				<g:each in="${surveyPage.getSections(program)}" var="section">
 					<g:set var="enteredSection" value="${surveyPage.enteredSections[section]}"/>
 
-					<li id="section-${section.id}">
```
Comment (fterrier) [inline grails-app/views/survey/_menu.gsp:24]: wait i think that's needed in the JS for survey

File: grails-app/views/survey/_menu.gsp [hunk 2]
```diff
@@ -4,8 +4,8 @@
 	<g:each in="${surveyPage.getPrograms()}" var="program">
 		<g:set var="enteredProgram" value="${surveyPage.enteredPrograms[program]}"/>
 		
-		<li id="program-${program.id}" class="foldable ${surveyPage.program?.id == program.id?'current':''}">
```
Comment (fterrier) [inline grails-app/views/survey/_menu.gsp:7]: same for this ID

File: grails-app/views/tags/filter/_locationTree.gsp [hunk 3]
```diff
@@ -14,7 +14,7 @@
 	<g:if test="${location.children != null && !location.children.empty}">		
 		<g:each in="${location.children}" var="child">
 			<g:if test="${locationFilterTree.contains(child)}">
-				<ul class="location-fold" id="location-fold-${location.id}">
```
Comment (fterrier) [inline grails-app/views/tags/filter/_locationTree.gsp:17]: is this ID not used anywhere ?

File: grails-app/views/tags/filter/_programTree.gsp [hunk 4]
```diff
@@ -15,7 +15,7 @@
 	<g:if test="${program.children != null || !program.children.empty}">					
 		<g:each in="${program.children}" var="child">
 			<g:if test="${programTree.contains(child)}">
-				<ul class="location-fold" id="location-fold-${program.id}">
```
Comment (fterrier) [inline grails-app/views/tags/filter/_programTree.gsp:18]: is this ID not used anywhere ?

**Comment by susanm74** [comment]
+1, we just talked about this on Skype, but with substantial style updates, it would be great to have a way to go through brute force and find orphan ids, classes, etc, and get rid of any unused code to keep things clean, and to have a way to verify that style/js ids, classes, etc are not getting mixed together. Redmine tix?

**Pull request merged**